fibsem/microscope.py

```diff
@@ -2619,6 +2627,60 @@ def draw_bitmap_pattern(self, pattern_settings: FibsemBitmapSettings):
         self._patterns.append(pattern)
         return pattern
 
+    def _resize_bitmap_to_pattern(
+        self, pattern_settings: FibsemBitmapSettings
+    ) -> NDArray[np.float_ | np.uint8]:
+        points = pattern_settings.bitmap
+
+        if points is None:
+            raise ValueError(
+                "Unable to resize bitmap as FibsemBitmapSettings.bitmap is None"
+            )
+
+        # Get pitch to calculate expected pixel size
+        rectangle = self.connection.patterning.create_rectangle(
+            center_x=pattern_settings.centre_x,
+            center_y=pattern_settings.centre_y,
+            width=pattern_settings.width,
+            height=pattern_settings.height,
+            depth=pattern_settings.depth,
+        )
+        pitch = rectangle.pitch_x
+        rectangle.enabled = False
+
+        new_shape = np.round(
```

Comment:
Any particular reason to cast this as an np.array rather than just Tuple[int, int]?